From 707d117c6ea8a073a79bde5b612c440849d771fb Mon Sep 17 00:00:00 2001
From: Frederic Bonnard <frediz@linux.vnet.ibm.com>
Date: Mon, 23 Mar 2015 19:23:54 +0100
Subject: [PATCH] Making urls relative

The goal is to be able to have kimchi website elsewhere than / , for example
https://server/kimchi/
---
 ui/css/theme-default/template_add.css |  20 ++--
 ui/css/theme-default/topbar.css       |   2 +-
 ui/js/src/kimchi.api.js               | 176 +++++++++++++++++-----------------
 ui/js/src/kimchi.guest_main.js        |   5 +-
 ui/js/src/kimchi.login.js             |   2 +-
 ui/js/src/kimchi.main.js              |   8 +-
 ui/pages/guest.html.tmpl              |   2 +-
 ui/pages/help/dita-help.xsl           |   4 +-
 ui/pages/kimchi-ui.html.tmpl          |   4 +-
 ui/pages/storagepool-add.html.tmpl    |   2 +-
 ui/pages/tabs/storage.html.tmpl       |   2 +-
 ui/pages/template-add.html.tmpl       |   2 +-
 12 files changed, 116 insertions(+), 113 deletions(-)

--- a/ui/css/theme-default/template_add.css
+++ b/ui/css/theme-default/template_add.css
@@ -41,7 +41,7 @@
     float: left; display : block;
     width: 50px;
     height: 52px;
-    background: url(../../images/theme-default/icon-back.png) center
+    background: url(../images/theme-default/icon-back.png) center
                 center no-repeat;
     cursor: pointer;
     display: block;
@@ -59,17 +59,17 @@
     margin: 0 10px 10px;
     padding: 20px 10px 20px 65px;
     border: 2px solid #ccc;
-    background: url(../../images/theme-default/icon-local.png) 15px
+    background: url(../images/theme-default/icon-local.png) 15px
                 center no-repeat;
     cursor: pointer;
 }
 
 .step-choose>li>a.local {
-    background-image: url(../../images/theme-default/icon-local.png);
+    background-image: url(../images/theme-default/icon-local.png);
 }
 
 .step-choose>li>a.remote {
-    background-image: url(../../images/theme-default/icon-remote.png);
+    background-image: url(../images/theme-default/icon-remote.png);
 }
 
 .step-choose>li>a:HOVER {
@@ -176,27 +176,27 @@
 }
 
 .iso-icon.centos {
-    background-image: url(../../images/icon-centos.png);
+    background-image: url(../images/icon-centos.png);
 }
 
 .iso-icon.debian {
-    background-image: url(../../images/icon-debian.png);
+    background-image: url(../images/icon-debian.png);
 }
 
 .iso-icon.fedora {
-    background-image: url(../../images/icon-fedora.png);
+    background-image: url(../images/icon-fedora.png);
 }
 
 .iso-icon.opensuse {
-    background-image: url(../../images/icon-opensuse.png);
+    background-image: url(../images/icon-opensuse.png);
 }
 
 .iso-icon.ubuntu {
-    background-image: url(../../images/icon-ubuntu.png);
+    background-image: url(../images/icon-ubuntu.png);
 }
 
 .iso-icon.gentoo {
-    background-image: url(../../images/icon-gentoo.png);
+    background-image: url(../images/icon-gentoo.png);
 }
 
 .list-iso {
--- a/ui/css/theme-default/topbar.css
+++ b/ui/css/theme-default/topbar.css
@@ -132,7 +132,7 @@
 }
 
 #user-icon {
-    background: url("/images/theme-default/user-icon.png") no-repeat left top;
+    background: url("../images/theme-default/user-icon.png") no-repeat left top;
     height: 16px;
     width: 16px;
 }
--- a/ui/js/src/kimchi.api.js
+++ b/ui/js/src/kimchi.api.js
@@ -54,7 +54,7 @@
     getCapabilities : function(suc, err, done) {
         done = typeof done !== 'undefined' ? done: function(){};
         kimchi.requestJSON({
-            url : "/config/capabilities",
+            url : "config/capabilities",
             type : "GET",
             contentType : "application/json",
             dataType : "json",
@@ -69,7 +69,7 @@
      */
     getI18n: function(suc, err, url, sync) {
         kimchi.requestJSON({
-            url : url ? url : kimchi.url + 'i18n.json',
+            url : url ? url : 'i18n.json',
             type : 'GET',
             resend: true,
             dataType : 'json',
@@ -84,7 +84,7 @@
      */
     getHost: function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'host',
+            url : 'host',
             type : 'GET',
             resend: true,
             contentType : 'application/json',
@@ -99,7 +99,7 @@
      */
     getHostStats : function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'host/stats',
+            url : 'host/stats',
             type : 'GET',
             contentType : 'application/json',
             headers: {'Kimchi-Robot': 'kimchi-robot'},
@@ -114,7 +114,7 @@
      */
     getHostStatsHistory : function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'host/stats/history',
+            url : 'host/stats/history',
             type : 'GET',
             resend: true,
             contentType : 'application/json',
@@ -138,7 +138,7 @@
      */
     createVM : function(settings, suc, err) {
         kimchi.requestJSON({
-            url : "/vms",
+            url : "vms",
             type : "POST",
             contentType : "application/json",
             data : JSON.stringify(settings),
@@ -154,7 +154,7 @@
      */
     createTemplate : function(settings, suc, err) {
         kimchi.requestJSON({
-            url : "/templates",
+            url : "templates",
             type : "POST",
             contentType : "application/json",
             data : JSON.stringify(settings),
@@ -166,7 +166,7 @@
 
     deleteTemplate : function(tem, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'templates/' + encodeURIComponent(tem),
+            url : 'templates/' + encodeURIComponent(tem),
             type : 'DELETE',
             contentType : 'application/json',
             dataType : 'json',
@@ -177,7 +177,7 @@
 
     cloneTemplate : function(tem, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'templates/' + encodeURIComponent(tem) + "/clone",
+            url : 'templates/' + encodeURIComponent(tem) + "/clone",
             type : 'POST',
             contentType : 'application/json',
             dataType : 'json',
@@ -188,7 +188,7 @@
 
     listTemplates : function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'templates',
+            url : 'templates',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -202,7 +202,7 @@
      */
     retrieveTemplate : function(templateName, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + "templates/" + encodeURIComponent(templateName),
+            url : "templates/" + encodeURIComponent(templateName),
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json'
@@ -216,7 +216,7 @@
      */
     updateTemplate : function(name, settings, suc, err) {
         $.ajax({
-            url : kimchi.url + "templates/" + encodeURIComponent(name),
+            url : "templates/" + encodeURIComponent(name),
             type : 'PUT',
             contentType : 'application/json',
             data : JSON.stringify(settings),
@@ -232,7 +232,7 @@
      */
     createStoragePool : function(settings, suc, err) {
         kimchi.requestJSON({
-            url : '/storagepools',
+            url : 'storagepools',
             type : 'POST',
             contentType : 'application/json',
             data : JSON.stringify(settings),
@@ -242,7 +242,7 @@
 
     updateStoragePool : function(name, content, suc, err) {
         $.ajax({
-            url : kimchi.url + "storagepools/" + encodeURIComponent(name),
+            url : "storagepools/" + encodeURIComponent(name),
             type : 'PUT',
             contentType : 'application/json',
             dataType : 'json',
@@ -254,7 +254,7 @@
 
     startVM : function(vm, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'vms/' + encodeURIComponent(vm) + '/start',
+            url : 'vms/' + encodeURIComponent(vm) + '/start',
             type : 'POST',
             contentType : 'application/json',
             dataType : 'json',
@@ -265,7 +265,7 @@
 
     poweroffVM : function(vm, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'vms/' + encodeURIComponent(vm) + '/poweroff',
+            url : 'vms/' + encodeURIComponent(vm) + '/poweroff',
             type : 'POST',
             contentType : 'application/json',
             dataType : 'json',
@@ -276,7 +276,7 @@
 
     shutdownVM : function(vm, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'vms/' + encodeURIComponent(vm) + '/shutdown',
+            url : 'vms/' + encodeURIComponent(vm) + '/shutdown',
             type : 'POST',
             contentType : 'application/json',
             dataType : 'json',
@@ -287,7 +287,7 @@
 
     resetVM : function(vm, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'vms/' + encodeURIComponent(vm) + '/reset',
+            url : 'vms/' + encodeURIComponent(vm) + '/reset',
             type : 'POST',
             contentType : 'application/json',
             dataType : 'json',
@@ -305,7 +305,7 @@
      */
     retrieveVM : function(vm, suc, err) {
         $.ajax({
-            url : kimchi.url + 'vms/' + encodeURIComponent(vm),
+            url : 'vms/' + encodeURIComponent(vm),
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -319,7 +319,7 @@
      */
     updateVM : function(name, settings, suc, err) {
         $.ajax({
-            url : kimchi.url + "vms/" + encodeURIComponent(name),
+            url : "vms/" + encodeURIComponent(name),
             type : 'PUT',
             contentType : 'application/json',
             data : JSON.stringify(settings),
@@ -331,7 +331,7 @@
 
     deleteVM : function(vm, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'vms/' + encodeURIComponent(vm),
+            url : 'vms/' + encodeURIComponent(vm),
             type : 'DELETE',
             contentType : 'application/json',
             dataType : 'json',
@@ -342,13 +342,13 @@
 
     vncToVM : function(vm) {
         kimchi.requestJSON({
-            url : '/config',
+            url : 'config',
             type : 'GET',
             dataType : 'json'
         }).done(function(data, textStatus, xhr) {
             proxy_port = data['display_proxy_port'];
             kimchi.requestJSON({
-                url : "/vms/" + encodeURIComponent(vm) + "/connect",
+                url : "vms/" + encodeURIComponent(vm) + "/connect",
                 type : "POST",
                 dataType : "json"
             }).done(function() {
@@ -374,13 +374,13 @@
 
     spiceToVM : function(vm) {
         kimchi.requestJSON({
-            url : '/config',
+            url : 'config',
             type : 'GET',
             dataType : 'json'
         }).done(function(data, textStatus, xhr) {
             proxy_port = data['display_proxy_port'];
             kimchi.requestJSON({
-                url : "/vms/" + encodeURIComponent(vm) + "/connect",
+                url : "vms/" + encodeURIComponent(vm) + "/connect",
                 type : "POST",
                 dataType : "json"
             }).done(function(data, textStatus, xhr) {
@@ -406,7 +406,7 @@
 
     listVMs : function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'vms',
+            url : 'vms',
             type : 'GET',
             contentType : 'application/json',
             headers: {'Kimchi-Robot': 'kimchi-robot'},
@@ -419,7 +419,7 @@
 
     listTemplates : function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'templates',
+            url : 'templates',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -431,7 +431,7 @@
 
     listStoragePools : function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'storagepools',
+            url : 'storagepools',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -443,7 +443,7 @@
 
     listStorageVolumes : function(poolName, suc, err) {
         $.ajax({
-            url : kimchi.url + 'storagepools/' + encodeURIComponent(poolName) + '/storagevolumes',
+            url : 'storagepools/' + encodeURIComponent(poolName) + '/storagevolumes',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -454,7 +454,7 @@
 
     listIsos : function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'storagepools/kimchi_isos/storagevolumes',
+            url : 'storagepools/kimchi_isos/storagevolumes',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -465,7 +465,7 @@
 
     listDistros : function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'config/distros',
+            url : 'config/distros',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -527,7 +527,7 @@
 
     getTask : function(taskId, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'tasks/' + encodeURIComponent(taskId),
+            url : 'tasks/' + encodeURIComponent(taskId),
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -538,7 +538,7 @@
 
     getTasksByFilter : function(filter, suc, err, sync) {
         kimchi.requestJSON({
-            url : kimchi.url + 'tasks?' + filter,
+            url : 'tasks?' + filter,
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -550,7 +550,7 @@
 
     login : function(settings, suc, err) {
         $.ajax({
-            url : "/login",
+            url : "login",
             type : "POST",
             contentType : "application/json",
             data : JSON.stringify(settings),
@@ -560,7 +560,7 @@
 
     logout : function(suc, err) {
         kimchi.requestJSON({
-            url : '/logout',
+            url : 'logout',
             type : 'POST',
             contentType : "application/json",
             dataType : "json"
@@ -569,7 +569,7 @@
 
     deleteStoragePool : function(poolName, suc, err) {
         $.ajax({
-            url : kimchi.url + 'storagepools/' + encodeURIComponent(poolName),
+            url : 'storagepools/' + encodeURIComponent(poolName),
             type : 'DELETE',
             contentType : 'application/json',
             dataType : 'json',
@@ -581,7 +581,7 @@
     changePoolState : function(poolName, state, suc, err) {
         if (state === 'activate' || state === 'deactivate')
             $.ajax({
-                url : kimchi.url + 'storagepools/' + encodeURIComponent(poolName) + '/' + state,
+                url : 'storagepools/' + encodeURIComponent(poolName) + '/' + state,
                 type : 'POST',
                 contentType : 'application/json',
                 dataType : 'json',
@@ -592,7 +592,7 @@
 
     listPlugins : function(suc, err, sync) {
         kimchi.requestJSON({
-            url : kimchi.url + 'plugins',
+            url : 'plugins',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -605,7 +605,7 @@
 
     listNetworks : function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'networks',
+            url : 'networks',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -620,7 +620,7 @@
     toggleNetwork : function(name, on, suc, err) {
         var action = on ? "activate" : "deactivate";
         kimchi.requestJSON({
-            url : kimchi.url + 'networks/' + encodeURIComponent(name) + '/' + action,
+            url : 'networks/' + encodeURIComponent(name) + '/' + action,
             type : 'POST',
             contentType : 'application/json',
             dataType : 'json',
@@ -633,7 +633,7 @@
 
     createNetwork : function(network, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'networks',
+            url : 'networks',
             type : 'POST',
             contentType : 'application/json',
             dataType : 'json',
@@ -647,7 +647,7 @@
 
     getInterfaces : function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'interfaces',
+            url : 'interfaces',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -661,7 +661,7 @@
 
     deleteNetwork : function(name, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'networks/' + encodeURIComponent(name),
+            url : 'networks/' + encodeURIComponent(name),
             type : 'DELETE',
             contentType : 'application/json',
             dataType : 'json',
@@ -674,7 +674,7 @@
 
     listReports : function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'debugreports',
+            url : 'debugreports',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -717,7 +717,7 @@
         };
 
         kimchi.requestJSON({
-            url : kimchi.url + 'debugreports',
+            url : 'debugreports',
             type : "POST",
             contentType : "application/json",
             data : JSON.stringify(settings),
@@ -729,7 +729,7 @@
 
     renameReport : function(name, settings, suc, err) {
         $.ajax({
-            url : kimchi.url + "debugreports/" + encodeURIComponent(name),
+            url : "debugreports/" + encodeURIComponent(name),
             type : 'PUT',
             contentType : 'application/json',
             data : JSON.stringify(settings),
@@ -742,7 +742,7 @@
     deleteReport: function(settings, suc, err) {
         var reportName = encodeURIComponent(settings['name']);
         kimchi.requestJSON({
-            url : kimchi.url + 'debugreports/' + reportName,
+            url : 'debugreports/' + reportName,
             type : 'DELETE',
             contentType : 'application/json',
             dataType : 'json',
@@ -757,7 +757,7 @@
 
     shutdown: function(settings, suc, err) {
         var reboot = settings && settings['reboot'] === true;
-        var url = kimchi.url + 'host/' + (reboot ? 'reboot' : 'shutdown');
+        var url = 'host/' + (reboot ? 'reboot' : 'shutdown');
         kimchi.requestJSON({
             url : url,
             type : 'POST',
@@ -770,7 +770,7 @@
 
     listHostPartitions : function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'host/partitions',
+            url : 'host/partitions',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -780,7 +780,7 @@
     },
 
     getStorageServers: function(type, suc, err) {
-        var url = kimchi.url + 'storageservers?_target_type=' + type;
+        var url = 'storageservers?_target_type=' + type;
         kimchi.requestJSON({
             url : url,
             type : 'GET',
@@ -794,7 +794,7 @@
     },
 
     getStorageTargets: function(server,type, suc, err) {
-        var url = kimchi.url + 'storageservers/' + server + '/storagetargets?_target_type=' + type;
+        var url = 'storageservers/' + server + '/storagetargets?_target_type=' + type;
         kimchi.requestJSON({
             url : url,
             type : 'GET',
@@ -820,7 +820,7 @@
     },
 
     getStoragePoolVolume: function(poolName, volumeName, suc, err) {
-        var url = kimchi.url + 'storagepools/' + encodeURIComponent(poolName) + '/storagevolumes/' + encodeURIComponent(volumeName);
+        var url = 'storagepools/' + encodeURIComponent(poolName) + '/storagevolumes/' + encodeURIComponent(volumeName);
         kimchi.requestJSON({
             url : url,
             type : 'GET',
@@ -836,7 +836,7 @@
         var vm = encodeURIComponent(settings['vm']);
         delete settings['vm'];
         kimchi.requestJSON({
-            url : kimchi.url + 'vms/' + vm + '/storages',
+            url : 'vms/' + vm + '/storages',
             type : 'POST',
             contentType : 'application/json',
             data : JSON.stringify(settings),
@@ -850,7 +850,7 @@
         var vm = encodeURIComponent(settings['vm']);
         var dev = encodeURIComponent(settings['dev']);
         kimchi.requestJSON({
-            url : kimchi.url + "vms/" + vm + '/storages/' + dev,
+            url : "vms/" + vm + '/storages/' + dev,
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -863,7 +863,7 @@
         var vm = encodeURIComponent(settings['vm']);
         var dev = encodeURIComponent(settings['dev']);
         kimchi.requestJSON({
-            url : kimchi.url + 'vms/' + vm + '/storages/' + dev,
+            url : 'vms/' + vm + '/storages/' + dev,
             type : 'PUT',
             contentType : 'application/json',
             data : JSON.stringify({
@@ -879,7 +879,7 @@
         var vm = settings['vm'];
         var dev = settings['dev'];
         kimchi.requestJSON({
-            url : kimchi.url + 'vms/' + encodeURIComponent(vm) +
+            url : 'vms/' + encodeURIComponent(vm) +
                       '/storages/' + encodeURIComponent(dev),
             type : 'DELETE',
             contentType : 'application/json',
@@ -892,7 +892,7 @@
     listVMStorages : function(params, suc, err) {
         var vm = encodeURIComponent(params['vm']);
         var type = params['storageType'];
-        var url = kimchi.url + 'vms/' + vm + '/storages';
+        var url = 'vms/' + vm + '/storages';
         if(type) {
             url += '?type=' + type;
         }
@@ -908,7 +908,7 @@
 
     listSoftwareUpdates : function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'host/packagesupdate',
+            url : 'host/packagesupdate',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -948,7 +948,7 @@
         };
 
         kimchi.requestJSON({
-            url : kimchi.url + 'host/swupdate',
+            url : 'host/swupdate',
             type : "POST",
             contentType : "application/json",
             dataType : "json",
@@ -972,7 +972,7 @@
     retrieveRepository : function(repository, suc, err) {
         var reposID = encodeURIComponent(repository);
         kimchi.requestJSON({
-            url : kimchi.url + "host/repositories/" + reposID,
+            url : "host/repositories/" + reposID,
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -984,7 +984,7 @@
     updateRepository : function(name, settings, suc, err) {
         var reposID = encodeURIComponent(name);
         $.ajax({
-            url : kimchi.url + "host/repositories/" + reposID,
+            url : "host/repositories/" + reposID,
             type : 'PUT',
             contentType : 'application/json',
             data : JSON.stringify(settings),
@@ -997,7 +997,7 @@
     enableRepository : function(name, enable, suc, err) {
         var reposID = encodeURIComponent(name);
         $.ajax({
-            url : kimchi.url + "host/repositories/" + reposID +
+            url : "host/repositories/" + reposID +
                 '/' + (enable === true ? 'enable' : 'disable'),
             type : 'POST',
             contentType : 'application/json',
@@ -1010,7 +1010,7 @@
     deleteRepository : function(repository, suc, err) {
         var reposID = encodeURIComponent(repository);
         kimchi.requestJSON({
-            url : kimchi.url + 'host/repositories/' + reposID,
+            url : 'host/repositories/' + reposID,
             type : 'DELETE',
             contentType : 'application/json',
             dataType : 'json',
@@ -1021,7 +1021,7 @@
 
     listRepositories : function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'host/repositories',
+            url : 'host/repositories',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -1032,7 +1032,7 @@
     },
 
     getHostFCDevices: function(suc, err) {
-        var url = kimchi.url+'host/devices?_cap=fc_host';
+        var url = 'host/devices?_cap=fc_host';
         kimchi.requestJSON({
             url : url,
             type : 'GET',
@@ -1046,7 +1046,7 @@
     },
 
     getGuestInterfaces: function(name, suc, err) {
-        var url = kimchi.url+'vms/'+encodeURIComponent(name)+'/ifaces';
+        var url = 'vms/'+encodeURIComponent(name)+'/ifaces';
         kimchi.requestJSON({
             url : url,
             type : 'GET',
@@ -1061,7 +1061,7 @@
 
     createGuestInterface : function(name, interface, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url+'vms/'+encodeURIComponent(name)+'/ifaces',
+            url : 'vms/'+encodeURIComponent(name)+'/ifaces',
             type : 'POST',
             contentType : 'application/json',
             dataType : 'json',
@@ -1075,7 +1075,7 @@
 
     deleteGuestInterface : function(vm, mac, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url+'vms/'+encodeURIComponent(vm)+'/ifaces/'+encodeURIComponent(mac),
+            url : 'vms/'+encodeURIComponent(vm)+'/ifaces/'+encodeURIComponent(mac),
             type : 'DELETE',
             contentType : 'application/json',
             dataType : 'json',
@@ -1088,7 +1088,7 @@
 
     updateGuestInterface : function(vm, mac, interface, suc, err) {
         $.ajax({
-            url : kimchi.url+'vms/'+encodeURIComponent(vm)+'/ifaces/'+encodeURIComponent(mac),
+            url : 'vms/'+encodeURIComponent(vm)+'/ifaces/'+encodeURIComponent(mac),
             type : 'PUT',
             contentType : 'application/json',
             data : JSON.stringify(interface),
@@ -1102,7 +1102,7 @@
 
     getUserById : function(data, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'users?_user_id=' + data.user_id,
+            url : 'users?_user_id=' + data.user_id,
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -1117,7 +1117,7 @@
 
     getUsers : function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'users',
+            url : 'users',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -1131,7 +1131,7 @@
 
     getGroups : function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'groups',
+            url : 'groups',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -1145,7 +1145,7 @@
 
     getHostPCIDevices : function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'host/devices?_passthrough=true&_cap=pci',
+            url : 'host/devices?_passthrough=true&_cap=pci',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -1159,7 +1159,7 @@
 
     getPCIDeviceCompanions : function(pcidev, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'host/devices?_passthrough_affected_by=' + pcidev,
+            url : 'host/devices?_passthrough_affected_by=' + pcidev,
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -1175,7 +1175,7 @@
         server = encodeURIComponent(server);
         port = port ? '&_server_port='+encodeURIComponent(port) : '';
         kimchi.requestJSON({
-            url : kimchi.url + 'storageservers/'+server+'/storagetargets?_target_type=iscsi'+port,
+            url : 'storageservers/'+server+'/storagetargets?_target_type=iscsi'+port,
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -1189,7 +1189,7 @@
 
     getPeers : function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'peers',
+            url : 'peers',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -1203,7 +1203,7 @@
 
     getVMPCIDevices : function(id, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'vms/'+encodeURIComponent(id)+'/hostdevs',
+            url : 'vms/'+encodeURIComponent(id)+'/hostdevs',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -1217,7 +1217,7 @@
 
     addVMPCIDevice : function(vm, device, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'vms/'+ encodeURIComponent(vm) +'/hostdevs',
+            url : 'vms/'+ encodeURIComponent(vm) +'/hostdevs',
             type : 'POST',
             contentType : 'application/json',
             dataType : 'json',
@@ -1231,7 +1231,7 @@
 
     removeVMPCIDevice : function(vm, device, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'vms/'+ encodeURIComponent(vm) +'/hostdevs/' + encodeURIComponent(device),
+            url : 'vms/'+ encodeURIComponent(vm) +'/hostdevs/' + encodeURIComponent(device),
             type : 'DELETE',
             contentType : 'application/json',
             dataType : 'json',
@@ -1249,7 +1249,7 @@
         var fd = settings['formData'];
         var sp = encodeURIComponent(settings['sp']);
         kimchi.requestJSON({
-            url : kimchi.url + 'storagepools/' + sp + '/storagevolumes',
+            url : 'storagepools/' + sp + '/storagevolumes',
             type : 'POST',
             data : fd,
             processData : false,
@@ -1267,7 +1267,7 @@
         var sp = encodeURIComponent(settings['sp']);
         delete settings['sp'];
         kimchi.requestJSON({
-            url : kimchi.url + 'storagepools/' + sp + '/storagevolumes',
+            url : 'storagepools/' + sp + '/storagevolumes',
             type : 'POST',
             data : JSON.stringify(settings),
             contentType : 'application/json',
@@ -1279,7 +1279,7 @@
 
     cloneGuest: function(vm, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'vms/'+encodeURIComponent(vm)+"/clone",
+            url : 'vms/'+encodeURIComponent(vm)+"/clone",
             type : 'POST',
             contentType : 'application/json',
             dataType : 'json',
@@ -1292,7 +1292,7 @@
 
     listSnapshots : function(vm, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url+'vms/'+encodeURIComponent(vm)+'/snapshots',
+            url : 'vms/'+encodeURIComponent(vm)+'/snapshots',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -1306,7 +1306,7 @@
 
     getCurrentSnapshot : function(vm, suc, err, sync) {
         kimchi.requestJSON({
-            url : kimchi.url+'vms/'+encodeURIComponent(vm)+'/snapshots/current',
+            url : 'vms/'+encodeURIComponent(vm)+'/snapshots/current',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
@@ -1321,7 +1321,7 @@
 
     revertSnapshot : function(vm, snapshot, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url+'vms/'+encodeURIComponent(vm)+'/snapshots/'+encodeURIComponent(snapshot)+'/revert',
+            url : 'vms/'+encodeURIComponent(vm)+'/snapshots/'+encodeURIComponent(snapshot)+'/revert',
             type : 'POST',
             contentType : 'application/json',
             dataType : 'json',
@@ -1334,7 +1334,7 @@
 
     createSnapshot : function(vm, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url+'vms/'+encodeURIComponent(vm)+'/snapshots',
+            url : 'vms/'+encodeURIComponent(vm)+'/snapshots',
             type : 'POST',
             contentType : 'application/json',
             dataType : 'json',
@@ -1347,7 +1347,7 @@
 
     deleteSnapshot : function(vm, snapshot, suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url+'vms/'+encodeURIComponent(vm)+'/snapshots/'+encodeURIComponent(snapshot),
+            url : 'vms/'+encodeURIComponent(vm)+'/snapshots/'+encodeURIComponent(snapshot),
             type : 'DELETE',
             contentType : 'application/json',
             dataType : 'json',
@@ -1360,7 +1360,7 @@
 
     getCPUInfo : function(suc, err) {
         kimchi.requestJSON({
-            url : kimchi.url + 'host/cpuinfo',
+            url : 'host/cpuinfo',
             type : 'GET',
             contentType : 'application/json',
             dataType : 'json',
--- a/ui/js/src/kimchi.guest_main.js
+++ b/ui/js/src/kimchi.guest_main.js
@@ -261,7 +261,10 @@
     guestTitle.html(vmObject.name);
 
     //Setup the VM console thumbnail display
-    var curImg = vmObject.screenshot || vmObject.icon;
+    var curImg = vmObject.icon;
+    if (vmObject.screenshot) {
+        curImg = vmObject.screenshot.replace(/^\//,'');
+    }
     var load_src = curImg || 'images/icon-vm.png';
     var tile_src = prevScreenImage || vmObject['load-src'];
     var liveTile=result.find('div[name=guest-tile] > .tile');
--- a/ui/js/src/kimchi.login.js
+++ b/ui/js/src/kimchi.login.js
@@ -57,7 +57,7 @@
                 var next_url = lastPage ? lastPage.replace(/\"/g,'') : "/";
             }
             kimchi.cookie.set('roles',JSON.stringify(data.roles));
-            window.location.replace(next_url)
+            window.location.replace(window.location.pathname.replace(/\/+login.html/, '') + next_url)
         }, function() {
             $("#messUserPass").show();
             $("#messSession").hide();
--- a/ui/js/src/kimchi.main.js
+++ b/ui/js/src/kimchi.main.js
@@ -90,8 +90,8 @@
         return tabs;
     };
 
-    var tabConfigUrl = '/config/ui/tabs.xml';
-    var pluginConfigUrl = '/plugins/{plugin}/ui/config/tab-ext.xml';
+    var tabConfigUrl = 'config/ui/tabs.xml';
+    var pluginConfigUrl = 'plugins/{plugin}/ui/config/tab-ext.xml';
     var pluginI18nUrl = 'plugins/{plugin}/i18n.json';
     var DEFAULT_HASH;
     var buildTabs = function(callback) {
@@ -344,9 +344,9 @@
     var url = ""
     // Find help page path according to tab name
     if (/^tabs/.test(path))
-        url = path.replace("tabs", "/help/" + lang);
+        url = path.replace("tabs", "help/" + lang);
     else if (/^plugins/.test(path))
-        url = path.slice(0, path.lastIndexOf('/')) + "/help/" + lang + path.slice(path.lastIndexOf('/'));
+        url = path.slice(0, path.lastIndexOf('/')) + "help/" + lang + path.slice(path.lastIndexOf('/'));
     // Checking if help page exist.
     $.ajax({
         url: url,
--- a/ui/pages/guest.html.tmpl
+++ b/ui/pages/guest.html.tmpl
@@ -45,7 +45,7 @@
                 <div class="tile ">
                     <img class="imgactive" alt="" src="">
                     <img class="imgload" alt="" src="">
-                    <img class="overlay shutoff-hidden" alt="$_("Start")" src="/images/theme-default/icon-power-down.png" >
+                    <img class="overlay shutoff-hidden" alt="$_("Start")" src="images/theme-default/icon-power-down.png" >
                 </div>
             </div>
            <div class="sortable guest-actions" name="guest-actions">
--- a/ui/pages/help/dita-help.xsl
+++ b/ui/pages/help/dita-help.xsl
@@ -9,8 +9,8 @@
             <head>
                 <title><xsl:value-of select="/cshelp/title" /></title>
                 <meta charset="UTF-8" />
-                <link rel="shortcut icon" href="/images/logo.ico" />
-                <link rel="stylesheet" type="text/css" href="/help/kimchi.css" />
+                <link rel="shortcut icon" href="../../images/logo.ico" />
+                <link rel="stylesheet" type="text/css" href="../kimchi.css" />
             </head>
             <body>
                 <xsl:apply-templates select="//cshelp" />
--- a/ui/pages/kimchi-ui.html.tmpl
+++ b/ui/pages/kimchi-ui.html.tmpl
@@ -118,7 +118,7 @@
 <script id="about-tmpl" type="kimchi/template">
     <div class="window about-window">
         <header>
-            <h1 class="title"><img alt="Kimchi logo" src="/images/logo.ico"/> $_("About")</h1>
+            <h1 class="title"><img alt="Kimchi logo" src="images/logo.ico"/> $_("About")</h1>
             <div class="close">X</div>
         </header>
         <div class="content" align="center">
@@ -126,7 +126,7 @@
                 <div>
                     <br/>
                     <br/>
-                    <img src="/images/theme-default/logo-plain.gif"/>
+                    <img src="images/theme-default/logo-plain.gif"/>
                     <br/>
                     <br/>
                     <h2>Kimchi</h2>
--- a/ui/pages/storagepool-add.html.tmpl
+++ b/ui/pages/storagepool-add.html.tmpl
@@ -100,7 +100,7 @@
                         <div class="host-partition">
                             <p class="text-help">
                                 $_("Looking for available partitions ...")
-                                <img src = "../images/theme-default/loading.gif" />
+                                <img src = "images/theme-default/loading.gif" />
                             </p>
                         </div>
                     </section>
--- a/ui/pages/tabs/storage.html.tmpl
+++ b/ui/pages/tabs/storage.html.tmpl
@@ -43,7 +43,7 @@
 </div>
 <div id="logicalPoolExtend" title="$_("Device path")">
      <p id="loading-info" class="text-help">
-         <img src = "../images/theme-default/loading.gif" />
+         <img src = "images/theme-default/loading.gif" />
          $_("Looking for available partitions ...")
      </p>
      <div class="host-partition">
--- a/ui/pages/template-add.html.tmpl
+++ b/ui/pages/template-add.html.tmpl
@@ -147,7 +147,7 @@
                 <div id="load-remote-iso">
                     <h3 class="step-subtitle">
                         <label>
-                            <img src = "../images/theme-default/loading.gif" />
+                            <img src = "images/theme-default/loading.gif" />
                             $_("Loading default remote ISOs ...")
                         </label>
                     </h3>
